### ISSUE TITLE
High CPU usage with Binance Electron app on Mac, source code not public

### ISSUE THREAD

**Issue by mo-bay** [binance/desktop#33]
Description:

I'm experiencing a problem with the Binance Electron app on my Mac running OS X 12.5, where it's using an unusually high amount of CPU, sometimes going up to 200%. This issue has been happening consistently for a few days now and it's causing my Mac to slow down significantly. I'm using Binance version 1.44.0.

It's worth noting that the Binance Electron app's source code is not public, which makes it difficult for users to determine what may be causing the high CPU usage.
Steps to Reproduce:

    Open the Binance Electron app on Mac
    Observe the CPU usage in Activity Monitor or Task Manager
    Notice that the app is consistently using a high amount of CPU, even when idle

Expected Behavior:

The Binance Electron app should use a reasonable amount of CPU when in use and idle, without causing the Mac to slow down significantly.
Actual Behavior:

The Binance Electron app is using an unusually high amount of CPU, causing the Mac to slow down significantly.

**Comment by zerobyted** [comment]
same for me, osx 10.15.5, Binance Helper (Renderer) boiling laptop.
I suppose it more related to how chrome is working, not specific app

**Comment by cesarvarela** [comment]
Same on windows, I'm really wondering if they mine crypto in the background,

**Comment by carlspring** [comment]
The OS is irrelevant here. I'm using Ubuntu 22.10 and I'm experiencing the very same issue.

I have a very high spec laptop with:
- 64 GB RAM
- 4 GB NVIDIA GeForce RTX 3050 Ti Laptop GPU (2560 cores)
- 12th Gen Intel(R) Core(TM) i9-12900HK @ 2900.000 MHz
- 4 TB SSD

I am running Binance with the "Discrete card" option and it's not using up all my GPU resources or anything like that:

```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.85.05    Driver Version: 525.85.05    CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ...  Off  | 00000000:01:00.0 Off |                  N/A |
| N/A   78C    P0    13W /  35W |   1656MiB /  4096MiB |     51%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|    0   N/A  N/A      2684      G   /usr/lib/xorg/Xorg                924MiB |
|    0   N/A  N/A      3038      G   /usr/bin/gnome-shell              730MiB |
|    0   N/A  N/A      7582      G   /usr/bin/nvidia-settings            0MiB |
+-----------------------------------------------------------------------------+
```

I have the most-recent NVidia driver:
![image](https://user-images.githubusercontent.com/1436265/226474593-b6413077-cb5a-495a-8f83-5e64f6ff30c2.png)

I don't believe the issue is with the video driver, because I'm not seeing such slowing down issues when playing games in Steam with Proton.

This is the only application that immediately sets off all of my laptop's fan to their max and I get 100C temperature readings. My machines becomes gradually slower and slower to a point where I need to restart it for it to make any sense to continue working. Often times this happens close to when I actually need to make a decision on a trade which is not cool, given the fact that this is crypro we're dealing with!

My suspicion is that this is coming from the Electron option that enables the crashpad. Could you please provide an option to switch this off so that we can try it out and see if it helps?

**Comment by steve-todorov** [comment]
There are a few things I've noticed that might cause this (although I don't really know the source code):

1. Binance client has some weird requests to a domain that is not verified as Binance's:
    ```
    $ dig +noall +answer www.yingwangtech.net @ns3.dnsv2.com
    www.yingwangtech.net.    600    IN    CNAME    u.yingwangtech.net.
    u.yingwangtech.net.      600    IN    CNAME    ga-bp1gv3ylf80yimr9v2kag.aliyunga0015.com.
    ga-bp1gv3ylf80yimr9v2kag.aliyunga0015.com => points to nothing right now?
    ```
    ![image](https://user-images.githubusercontent.com/2303663/226716419-01da7b97-a54d-46bc-9f48-f124d0f28dd7.png)

    This company seems to have been [registered in the UK](https://find-and-update.company-information.service.gov.uk/company/08310688/officers) and has been dissolved in 2014. 

    The domain is listed together with `yingwangtech.mobi` as malware in https://github.com/badmojr/1Hosts (specifically [this file](https://github.com/badmojr/1Hosts/blob/398b005e6c026948822dd71cdb713031475aeb0f/Xtra/hosts.txt))


2. Binance uses [Sentry](https://sentry.io/welcome/) to record the user session state and interactions (where you click and what happens when you click). Sentry is essentially doing screenshots N times a second and then submitting those into sentry.io which can be replayed. This could be one of the bottlenecks causing high CPU.

3. Maybe there is some malware that mines crypto that might have been introduced by a 3rd party dependency.


I am also affected by this issue and have been digging around it for a while.

**Comment by dhiyaulhaqZA** [comment]
Every time I open up my Binance app, it always call this url:
```
URL: https://47.101.151.69/resolve?name=www.yingwangtech.mobi&type=1&uid=1&key=****&pf=ios&sv=1.0.8&ts=1688788182&ak=****
```

When I search the IP 47.101.151.69, it is associated with Alibaba. I don't know what is this, is Binance associated with Alibaba?

**Comment by pawpawclaw** [comment]
> Every time I open up my Binance app, it always call this url:
> 
> ```
> URL: https://47.101.151.69/resolve?name=www.yingwangtech.mobi&type=1&uid=1&key=****&pf=ios&sv=1.0.8&ts=1688788182&ak=****
> ```
> 
> When I search the IP 47.101.151.69, it is associated with Alibaba. I don't know what is this, is Binance associated with Alibaba?

not exactly, these strange IP's are to ensure that users from within China can access the app without a vpn.
